javascript/URDFLoader.js

```diff
@@ -515,33 +541,18 @@ class URDFLoader {
                 rpy = this._processTuple(n.getAttribute('rpy'));
 
             } else if (type === 'material') {
-
-                this.forEach(n.children, c => {
-
-                    if (c.nodeName.toLowerCase() === 'color') {
-
-                        const rgba = c.getAttribute('rgba')
-                            .split(/\s/g)
-                            .map(v => parseFloat(v));
-
-                        material.color.r = rgba[0];
-                        material.color.g = rgba[1];
-                        material.color.b = rgba[2];
-                        material.opacity = rgba[3];
-
-                        if (material.opacity < 1) material.transparent = true;
-
-                    } else if (c.nodeName.toLowerCase() === 'texture') {
-
-                        const filename = c.getAttribute('filename').replace(/^(package:\/\/)/, '');
-                        const filePath = this._resolvePackagePath(packages, filename, path);
-
-                        material.map = this._textureloader.load(filePath);
-
-                    }
-
-                });
-
+                if (n.children.length) {
+                    this.forEach(n.children, c => {
+                        if (c.nodeName.toLowerCase() === 'color') {
+                            this._processMaterial(material, 'color', c.getAttribute('rgba'));
+                        } else if (c.nodeName.toLowerCase() === 'texture') {
+                            this._processMaterial(material, 'texture', c.getAttribute('filename'), packages, path);
+                        }
+                    });
+                } else {
+                    const name = n.getAttribute('name');
+                    this._processMaterial(material, materialMap[name].type, materialMap[name].value, packages, path);
```

Comment:
I see that the `_processMaterial` function takes an existing material and modifies it so each mesh has its own material instance. Considering these are links that share material nodes, does it make sense to preprocess material nodes into `THREE.Material` objects up front and share them across many `THREE.Mesh` instances? That way fewer material objects will be created. This also means that the meshes will share any material updates.

I'm also noticing we will always load a new texture when processing a material, even if that texture has been used elsewhere, which is maybe the bigger issue, memory-wise. I'll create a separate issue for that, though.